app/services/nomis/api_slot_correcter.rb

```diff
@@ -0,0 +1,35 @@
+module Nomis
+  class ApiSlotCorrecter
+    def initialize(raw_slot)
+      @raw_slot = ConcreteSlot.parse(raw_slot)
+    end
+
+    def slot
+      return @raw_slot if valid_times?
+
+      ConcreteSlot.parse_times(fix_time(@raw_slot.begin_at),
+        fix_time(@raw_slot.end_at))
+    end
+
+  private
+
+    def fix_time(time)
```

Comment:
Maybe ```#normalise_times``` would be a better name as ```fix``` suggests what comes back from the API is broken. But I'm not too fussed about it.